README.md

```diff
@@ -31,3 +24,17 @@ Run Java with the agent added as a JVM argument:
 
 Alternatively, if modifying the Java command line is not possible, the
 above may be added to the `JAVA_TOOL_OPTIONS` environment variable.
+
+# Alternatives
+
+Newer JVMs (8u92+) support the `-XX:ExitOnOutOfMemoryError` option to
```

Comment:
👍

Reply:
I've tried to use jvmkill inside docker cotainers, but killing the process (PID:1) inside the container didn't work even though the jvmkill programm was actually calling kill() system call. See also https://unix.stackexchange.com/questions/457649/unable-to-kill-process-with-pid-1-in-docker-container.

In this case, using `-XX:ExitOnOUtOfMemoryError` is the only option.

Reply:
It seems using `docker run --init ... ` is a workaround to make jvmkill work https://devops.stackexchange.com/questions/5613/how-to-explicitly-kill-the-process-with-pid-1-from-inside-a-container

Reply:
Interesting. Maybe we should have it call exit and then kill (just in case)?

Reply:
The recent version of docker seems supporting killing PID:1 just by using `docker run --init`. I'll try it.

Reply:
Yeah. `docker run --init` worked as expected. We should document that.

Reply:
Added a note #4

Reply:
@xerial thanks for the info! I don't expect people to use `--init` always (at least not by default), so like @electrum 's idea to do both: exit and kill (even though i dont see a reason why exit should not work)

Reply:
If exit()’s implementation is just sending SIGTERM to the process, it will not be effective for PID 1 without `—init` option. I’m not sure about it, so we need to experiment it.

Reply:
It shouldn't be signal-based. There is no need to.

Reply:
I think calling `_exit()` should suffice. We should actually do `SIGKILL` first, then `_exit()` if the process still exists at that point. But that's follow up, and we already added the note for Docker, so I'm going to resolve this conversation.